picard/script.py

```diff
@@ -910,6 +910,143 @@ def func_is_video(parser):
+    _start = int(start_index) if start_index else None
+    _end = int(end_index) if end_index else None
+    return text[_start:_end]
+
+
+def func_get_multi_item(parser, multi, item_index, separator=MULTI_VALUED_JOINER):
+    """Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+
+    Arguments:
+        multi {multi-value} -- The multi-value from which the item is to be retrieved.
+        item_index {int} -- The zero-based index of the item to be retrieved.
+
+    Returns:
+        str -- Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+    """
+    if item_index:
+        _index = int(item_index.eval(parser))
+    else:
+        return ''
+    _multi = _get_multi_values(parser, multi, separator)
+    try:
+        return str(_multi[_index])
+    except IndexError:
+        return ''
+
+
+def func_foreach_multi_item(parser, multi, loop_code, _loop_value="_loop_value", _loop_count="_loop_count", separator=MULTI_VALUED_JOINER):
+    """Iterates over each element found in the specified multi-value variable, executing the specified code.
+    For each loop, the element value is first stored in the tag specified by _loop_value and the count is
+    stored in the tag specified by _loop_count.  This allows the element or count value to be accessed within
+    the code script.
+
+    Arguments:
+        multi {multi-value} -- The multi-value to be iterated.
+        loop_code {str} -- Script code to be processed on each iteration.
+
+    Keyword Arguments:
+        _loop_value {str} -- Tag name to use for the element for the current loop (default: {"_loop_value"})
+        _loop_count {str} -- Tag name to use for the count of the current loop (default: {"_loop_count"})
+    """
+    loop_count = 0
+    _multi = _get_multi_values(parser, multi, separator)
+    for value in _multi:
+        loop_count += 1
+        func_set(parser, _loop_count, str(loop_count))
+        func_set(parser, _loop_value, str(value))
+        loop_code.eval(parser)
+    func_unset(parser, _loop_count)
+    func_unset(parser, _loop_value)
+    return ''
+
+
+def func_for_loop(parser, name, start, end, increment, loop_code):
+    """Standard Python 'for' loop.
+
+    Sets the value of name to the start value (integer) and loops through the specified code until the value
+    of name is greater than or equal to the end value (integer).  At the end of each loop, the value of name is
+    incremented by the specified increment value (integer).
+
+    Arguments:
+        name {str} -- Tag name to use for the loop count.  This can be accessed within the code block.
+        start {int} -- The initial value for the loop counter.
+        end {int} -- The value at which the loop exits.
+        increment {int} -- The value by which the loop count variable is incremented for each loop.
+        loop_code {str} -- Script code to be processed on each iteration.
+    """
```

Comment:
This function may not be that useful, it looks over complicated.

Reply:
Yeah, I only included it because there is a `for()` loop in Python.  I agree that everything that the `for()` loop provides can be done using the `while()` loop so I've removed the `for()` loop from the code.

Reply:
> Yeah, I only included it because there is a for() loop in Python.

Wrong.  I've been doing a lot of work in PHP lately, so that must have been what I was thinking of.  In any event, it's gone now.

Reply:
I guess Python not having this is what made me frown on  the for loop :D